merger_sort.c

```diff
@@ -0,0 +1,98 @@
+#include <stdio.h>
+
+#define SIZE 8
+
+int temp[SIZE] = {0};
+
+void merge (int array[], int start_1, int end_1, int start_2, int end_2)
+{
+    int i = 0;
+    
+    // while there are elements in both subarrays 
+    while ((start_1 <= end_1) && (start_2 <= end_2))
+    {
+        // compare numbers at the start of the subarrays 
+        if (array[start_1] > array[start_2])
+        {
+            // append smaller to array 
+            temp[i] = array[start_2];
+            start_2++; 
+            i++;
+        }
+        else
+        {
+            temp[i] = array[start_1];
+            start_1++; 
+            i++;
+        }
+    }
+    
+    // while elements remain in subbarray 2
+    while ((start_1 > end_1) && (start_2 <= end_2))
+    {
+        // append elements to array 
+        temp[i] = array[start_2];
+        i++; 
+        start_2++;
+    }
+    
+    // while elements remain in subbarray 1
+    while ((start_2 > end_2) && (start_1 <= end_1))
+    {
+        // append elements to array 
+        temp[i] = array[start_1];
+        i++; 
+        start_1++;
+    }
+    
+    //copy sorted values from the temp back to original array 
+    for (i = 0; j < SIZE; i++)
+    {
+        array[i] = temp[i];
+    }
+    
+}
+
+void sort (int array[], int start, int end)
+{
+    if (end > start)
```

Comment:
`if (end > start + 1)` should be fine, one element is sorted by definition (though in reality this just means n extra steps, not too big a problem).